CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc

```diff
@@ -0,0 +1,126 @@
+/****************************************************************************
+ *
+ * This is a part of PPS offline software.
+ * Authors:
+ *   Edoardo Bossini
+ *   Piotr Maciej Cwiklicki
+ *   Laurent Forthomme
+ *
+ ****************************************************************************/
+
+#include "DQMServices/Core/interface/DQMGlobalEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSGeometry.h"
+#include "Geometry/Records/interface/VeryForwardRealGeometryRecord.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSDiamondDetId.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSDiamondRecHit.h"
+
+#include "CalibPPS/TimingCalibration/interface/TimingCalibrationStruct.h"
+
+//------------------------------------------------------------------------------
+
+class PPSTimingCalibrationPCLWorker : public DQMGlobalEDAnalyzer<TimingCalibrationHistograms> {
+public:
+  explicit PPSTimingCalibrationPCLWorker(const edm::ParameterSet&);
+
+  void dqmAnalyze(const edm::Event&, const edm::EventSetup&, const TimingCalibrationHistograms&) const override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  void bookHistograms(DQMStore::IBooker&,
+                      const edm::Run&,
+                      const edm::EventSetup&,
+                      TimingCalibrationHistograms&) const override;
+
+  edm::EDGetTokenT<edm::DetSetVector<CTPPSDiamondRecHit>> diamondRecHitToken_;
+  edm::ESGetToken<CTPPSGeometry, VeryForwardRealGeometryRecord> geomEsToken_;
+
+  const std::string dqmDir_;
+};
+
+//------------------------------------------------------------------------------
+
+PPSTimingCalibrationPCLWorker::PPSTimingCalibrationPCLWorker(const edm::ParameterSet& iConfig)
+    : diamondRecHitToken_(
+          consumes<edm::DetSetVector<CTPPSDiamondRecHit>>(iConfig.getParameter<edm::InputTag>("diamondRecHitTag"))),
+      geomEsToken_(esConsumes()),
```

Comment:
The consumes declaration needs the transition type because the token is used in non-Event transition
```suggestion
      geomEsToken_(esConsumes<edm::Transition::BeginRun>()),
```